src/sql/workbench/contrib/notebook/test/electron-browser/notebookModel.test.ts

```diff
@@ -205,6 +208,163 @@ suite('notebook model', function (): void {
 		assert.deepEqual(model.cells[1].source, expectedNotebookContent.cells[1].source);
 	});
 
+	test('Should handle multiple notebook managers', async function (): Promise<void> {
+		// Given a notebook with 2 cells
+		let mockContentManager = TypeMoq.Mock.ofType(NotebookEditorContentManager);
+		mockContentManager.setup(c => c.loadContent()).returns(() => Promise.resolve(expectedNotebookContent));
+		defaultModelOptions.contentManager = mockContentManager.object;
+
+		let defaultNotebookManager = new NotebookManagerStub();
+		defaultNotebookManager.providerId = 'SQL';
+
+		let jupyterNotebookManager = new NotebookManagerStub();
+		jupyterNotebookManager.providerId = 'jupyter';
+
+		// Setup 2 notebook managers
+		defaultModelOptions.notebookManagers = [defaultNotebookManager, jupyterNotebookManager];
+
+		// Change default notebook provider id to jupyter
+		defaultModelOptions.providerId = 'jupyter';
+
+		// When I initalize the model
+		let model = new NotebookModel(defaultModelOptions, undefined, logService, undefined, undefined);
+		await model.loadContents();
+
+		// I expect the default provider to be jupyter
+		assert.equal(model.notebookManager.providerId, 'jupyter');
```

Comment:
For all these assertions it would be good to add a message explaining what went wrong so it's easy to track down which assert failed (because right now it'll just say something like `'SomeOtherProviderId did not equal 'jupyter'`

Reply:
Cool, will add.